diesel/src/query_source/joins.rs

```diff
@@ -180,3 +174,33 @@ impl<DB: Backend> QueryFragment<DB> for LeftOuter {
         Ok(())
     }
 }
+
+use super::{Succ, Never, AppearsInFromClause};
+
+impl<T, Left, Right, Kind> AppearsInFromClause<T> for Join<Left, Right, Kind> where
+    Left: AppearsInFromClause<T>,
+    Right: AppearsInFromClause<T>,
+    Left::Count: Plus<Right::Count>,
```

Comment:
I'm not sure I understand the question here? This is basically just saying that both counts are either `Never` or `Succ`.